manim/mobject/polyhedra.py

```diff
@@ -0,0 +1,361 @@
+    and the apex, the last coordinate in the list.
+
+    Next, we define the faces of the polyhedron. The triangular surfaces of the pyramid are polygons
+    with two adjacent vertices in the base and the vertex at the apex as corners. We thus define these
+    surfaces in the first four elements of our face list. The last element defines the base of the pyramid.
+
+    The graph and faces of polyhedra can also be accessed and modified directly, after instantiation.
+    They are stored in the `graph` and `faces` attributes respectively.
+
+    .. manim:: PolyhedronSubMobjects
+        :save_last_frame:
+
+        class PolyhedronSubMobjects(ThreeDScene):
+            def construct(self):
+                self.set_camera_orientation(phi=75 * DEGREES, theta=30 * DEGREES)
+                octahedron = Octahedron(edge_length = 3)
+                octahedron.graph[0].set_color(RED)
+                octahedron.faces[2].set_color(YELLOW)
+                self.add(octahedron)
+    """
+
+    def __init__(
+        self,
+        vertex_coords: List[np.ndarray],
+        faces_list: List[List[Hashable]],
+        faces_config: dict = {},
+        graph_config: dict = {},
+    ):
+        VGroup.__init__(self)
+        self.faces_config = dict(
+            {"fill_opacity": 0.5, "shade_in_3d": True}, **faces_config
+        )
+        self.graph_config = dict(
+            {
+                "vertex_type": Dot3D,
+                "edge_config": {
+                    "stroke_opacity": 0,  # I find that having the edges visible makes the polyhedra look weird
+                },
+            },
+            **graph_config
+        )
+        self.vertex_coords = vertex_coords
+        self.vertex_indices = list(range(len(self.vertex_coords)))
+        self.layout = dict(enumerate(self.vertex_coords))
+        self.faces_list = faces_list
+        self.face_coords = [[self.layout[j] for j in i] for i in faces_list]
+        self.edges = self.get_edges(self.faces_list)
+        self.faces = self.create_faces(self.face_coords)
+        self.graph = Graph(
+            self.vertex_indices, self.edges, layout=self.layout, **self.graph_config
+        )
+        self.add(self.faces, self.graph)
+        self.add_updater(self.update_faces)
+
+    def get_edges(self, faces_list):
+        """Creates list of cyclic pairwise tuples."""
+        edges = []
+        for face in faces_list:
+            edges += zip(face, face[1:] + face[:1])
+        return edges
+
+    def create_faces(self, face_coords):
+        """Creates VGroup of faces from a list of face coordinates."""
+        face_group = VGroup()
+        for face in face_coords:
+            face_group.add(Polygon(*face, **self.faces_config))
+        return face_group
+
+    def update_faces(self, m):
+        face_coords = self.extract_face_coords()
+        new_faces = self.create_faces(face_coords)
+        self.faces.match_points(new_faces)
+
+    def extract_face_coords(self):
```

Comment:
can you annotate the return type?

Reply:
How do I do that?

Reply:
See https://www.python.org/dev/peps/pep-3107/#return-values or the example in the starting of https://docs.python.org/3/library/typing.html

Reply:
Done